core/meta/src/TClass.cxx

```diff
@@ -3451,38 +3451,31 @@ const char *TClass::GetSharedLibs()
 
 TList *TClass::GetListOfBases()
 {
-   if (!fBase) {
-      if (fCanLoadClassInfo) {
```

Comment:
I added an early return, so the lock is taken much less often, but I agree that the commit message is not the best, so I will rewrite it and add a better explanation for the changes.